tests/integration/test_integration.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+import sys
+from pathlib import Path
+
+import pytest
+import torch
+import yaml
+from transformers.testing_utils import execute_subprocess_async
+
+from llm_studio.app_utils.utils import prepare_default_dataset
+
+
+def get_experiment_status(path: str) -> str:
+    """Get status information from experiment.
+
+    Args:
+        path: path to experiment folder
+    Returns:
+        experiment status
+    """
+
+    try:
+        flag_json_path = os.path.join(path, "flags.json")
+        if not os.path.exists(flag_json_path):
+            return "none"
+        with open(flag_json_path) as file:
+            flags = json.load(file)
+            status = flags.get("status", "none")
+        return status
+    except Exception:
+        return "none"
+
+
+@pytest.mark.parametrize(
+    "config_name",
+    [
+        "test_causal_language_modeling_oasst_cfg",
+        "test_sequence_to_sequence_modeling_oasst_cfg",
+        "test_rlhf_language_modeling_oasst_cfg",
+    ],
+)
+def test_oasst_training(tmp_path, config_name):
+    """
+    Test training on OASST dataset.
+
+    Pytest keeps around the last 3 test runs in the tmp_path fixture.
+    """
+
+    if torch.cuda.is_available():
```

Comment:
One could use 
`need_gpus = pytest.mark.skipif(not torch.cuda.is_available(), reason="GPU only test")`
`@need_gpus` here. Test would then be skipped completely.

Reply:
til :) 
thank you